script/check-posix-pty

```diff
@@ -140,25 +176,27 @@ send "coap resource TestResource\r\n"
 expect "Done"
 send "coap set TestContent\r\n"
 expect "Done"
-wait
+set timeout -1
+expect eof
 EOF
 
+    sleep 10
+
     # wait until the node becomes leader
-    while true; do
-        sleep 5
-        if grep -q leader $OT_OUTPUT; then
-            break
-        else
-            echo 'Still waiting for leader'
-        fi
-    done
+    timeout_run 5 wait_for_leader
 
     netstat -an | grep -q 5683 || die 'Application CoAP port is not available!'
```

Comment:
We start coap after node becomes leader. Is it possible for this check to occur before `coap start` is complete?

Reply:
added a sleep before.